Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -347,6 +346,45 @@ lemma gaussianReal_mul_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussian
 
 end Transformations
 
+open Measurable Real
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {p : Measure Ω} {μ : ℝ} {v : ℝ≥0} {X : Ω → ℝ}
+
+lemma mgf_dirac (hX : p.map X = .dirac μ) (t : ℝ) : mgf X p t = exp (μ * t) := by
+  rw [← mgf_id_map, mgf, hX, integral_dirac, mul_comm]
+  · rfl
+  · apply AEMeasurable.of_map_ne_zero
+    rw [hX]
+    exact IsProbabilityMeasure.ne_zero (.dirac μ)
```

Comment:
@YaelDillies , do you know how to condense this to one line?

Reply:
Here's two lines:
```lean
  have : IsProbabilityMeasure (p.map X) := by rw [hX]; infer_instance
  rw [← mgf_id_map (.of_map_ne_zero <| IsProbabilityMeasure.ne_zero _), mgf, hX, integral_dirac,
    mul_comm, id_def]
```
I wonder if there's a msising lemma here though to make this easier.